Refactoring

AleshaOleg opened the pull request.

AleshaOleg commented:
@kof please, review this pr after all others. It makes a lot of changes in file/structure and code style

kof commented on browsers.json:
```diff
@@ -1,67 +1,4 @@
 {
-  "BS_Android4.1": {
```
did you remove them by accident?

AleshaOleg replied:
yes, while testing:(

kof commented on package.json:
```diff
@@ -39,45 +39,46 @@
-    "eslint-plugin-import": "^1.12.0",
-    "eslint-plugin-jsx-a11y": "^1.5.5",
-    "eslint-plugin-react": "^5.0.1",
+    "caniuse-db": "^1.0.30000784",
```
why is it a direct dependency here?

AleshaOleg replied:
Because it should be always last version with fresh data and not depend on other packages (which can be use old version of caniuse-db)

kof replied:
We can have control over the other package using it and keep it up to date there. If you put it in here, we have to keep it up-to-date in both packages. This is a hack, we shouldn't use if not necesserily

AleshaOleg replied:
But what if they do not update this package to the latest one? I can't trust to others packages, and don't want too:)

kof replied:
who is they? it is you or me, otherwise you and me are gonna update it here and there. Don't create more jobs than we need.

kof commented on src/plugins/appearence.js:
```diff
@@ -0,0 +1,14 @@
+  noPrefill: ['appearance'],
+  supportedProperty: (prop) => {
+    if (prop === 'appearance') {
+      if (prefix.js === 'ms') {
```
If we are on IE, we prefix with -webkit? maybe worth a comment

AleshaOleg replied:
Please take a look to - https://caniuse.com/#search=appearance
Everything ok. Edge browser should be prefixed with `-webkit-`

kof replied:
I know, please add a comment "Microsoft Edge and IE Mobile support this property with the -webkit- prefix, rather than -ms- for interop reasons."

kof commented on src/plugins/appearence.js:
```diff
@@ -0,0 +1,14 @@
+export default {
+  noPrefill: ['appearance'],
+  supportedProperty: (prop) => {
+    if (prop === 'appearance') {
```
please use early return

AleshaOleg replied:
what you mean?

kof replied:
if (prop !== 'appearance') return false

AleshaOleg commented:
Tests fails because new `autoprefixer` (> 7, as I understood) which I added on this PR, using classes and Babel should to be properly plugins to resolve this case. I fixed this in autoprefixer, and send PR - https://github.com/postcss/autoprefixer/pull/961.

Should wait until new version of autoprefixer will be realeased

AleshaOleg commented:
`Autoprefixer` released support of IE 9/10 in 7.2.4 (https://github.com/postcss/autoprefixer/pull/963). So, I updated it to new one

kof commented on src/supported-value.js:
```diff
@@ -67,14 +78,21 @@ export default function supportedValue(property, value) {
 
+  if (isImportant) {
+    value += ' !important'
+    isImportant = false
```
why do you set it to false?

AleshaOleg replied:
Because every run of `supportedValue` function, we need `isImportant` to reset it (https://github.com/cssinjs/css-vendor/blob/08d2f32d671238df0ffd62d5be6f8e2524cca5d6/src/supported-value.js#L53), if `!important` exist in value

kof replied:
Omg just realized isImportant is outside of the function scope, why? Are you using it outside of that function? __This is a very dangerous side-effect style coding we should always avoid as far as possible__

kof commented on src/utils/camelize.js:
```diff
@@ -6,11 +6,11 @@ const regExp = /[-\s]+(.)?/g
 function toUpper(match, c) {
   return c ? c.toUpperCase() : ''
 }
-
```
we need last line always nl, there is an issue with this  https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Didn't linter warn about it? I was sure it is checking it.

AleshaOleg replied:
Hm, I didn't removed it. In addition, Atom add last line it automatically. I will check

kof replied:
you can make it always ensure last line, actually eslint should do that as well with --fix option

kof commented on src/plugins/break-props-old.js:
```diff
@@ -1,21 +1,20 @@
+      const jsProp = `WebkitColumn${pascalize(prop)}`
+      return jsProp in style ? `${prefix.css}column-${prop}` : false
+    }
+    else if (prefix.js === 'Moz') {
```
no need for else if there was a return

kof commented on src/supported-property.js:
```diff
@@ -42,25 +34,30 @@ function transitionTransformPrefix(options, prop) {
  * @return {String|Boolean}
  * @api public
  */
+const isPropertySupported = (prop) => {
```
please inline this, no need for a function
